routing/dht/routing.go

```diff
@@ -40,19 +40,24 @@ func (dht *IpfsDHT) PutValue(ctx context.Context, key u.Key, value []byte) error
 		return err
 	}
 
-	peers := dht.routingTable.NearestPeers(kb.ConvertKey(key), KValue)
-
-	query := newQuery(key, dht.network, func(ctx context.Context, p peer.ID) (*dhtQueryResult, error) {
```

Comment:
Ah, the reason i didnt use a query here is because my peer input was a channel, and not an array. Not sure how we should reconcile that difference. Thoughts?

Re: PutValue being incorrect, its not incorrect, it uses the getClosestPeers call. its *more* correct.